src/main/java/org/hiero/sketch/spreadsheet/Histogram1D.java

```diff
@@ -0,0 +1,88 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.hiero.sketch.table.api.IColumn;
+import org.hiero.sketch.table.api.IMembershipSet;
+import org.hiero.sketch.table.api.IRowIterator;
+import org.hiero.sketch.table.api.IStringConverter;
+
+public class Histogram1D implements IHistogram {
+    final private IColumn column;
+    final private IMembershipSet membershipSet;
+    final private double[] boundaries;
+    final private int[] buckets;
+    final private IStringConverter converter;
+    private int missingData;
+    private int outOfRange;
+
+    public Histogram1D(@NonNull IColumn column, @NonNull IMembershipSet membershipSet,
```

Comment:
This seems like it will accept an empty "boundaries" array when it shouldn't. isSorted will return true and then line 25 will complain about negative array sizes.